cmake/compiler_flags.cmake

```diff
@@ -29,8 +29,8 @@ else()
     endif()
 endif()
 
-# We need a define if on APPLE
 if(APPLE)
+    # We need a define if on APPLE
```

Comment:
Same thing here: `if(APPLE) add_definitions("-DAPPLE")` tells me that "if `APPLE` is defined, we add the definition called 'APPLE'". And I assume it is needed, otherwise we would not do it.

The comment represents 25% of this piece of code and just says exactly the same :P. Even worse: maybe at some point, the comment will get disconnected from the `add_definitions` call, and it will be misguiding.

Reply:
I guess I wrote these comments because I'm a cmake noob so for me it wasn't trivial that the variable `APPLE` would not result in a define `APPLE` while compiling. Now I know :smile: .

Reply:
Removed it :)